pkg/controllers/networkpolicy/networkpolicy_controller.go

```diff
@@ -227,6 +248,83 @@ func (r *NetworkPolicyReconciler) StartController(mgr ctrl.Manager, _ webhook.Se
 	return nil
 }
 
+// getAllPodPortNames extracts all named ports from a list of pods
+func getAllPodPortNames(pods []v1.Pod) sets.Set[string] {
+	podPortNames := sets.New[string]()
+	for _, pod := range pods {
+		for _, container := range pod.Spec.Containers {
+			for _, port := range container.Ports {
+				if port.Name != "" {
+					podPortNames.Insert(port.Name)
+				}
+			}
+		}
+	}
+	return podPortNames
+}
+
+// reconcileNetworkPolicy is triggered by Pod events or Namespace events to reconcile NetworkPolicies with named ports
+func reconcileNetworkPolicy(pkgClient client.Client, pods []v1.Pod, q workqueue.TypedRateLimitingInterface[reconcile.Request]) error {
+	podPortNames := getAllPodPortNames(pods)
+	log.V(1).Info("POD named port", "podPortNames", podPortNames)
+
+	npList := &networkingv1.NetworkPolicyList{}
+	err := pkgClient.List(context.Background(), npList)
+	if err != nil {
+		log.Error(err, "Failed to list all the network policies")
+		return err
+	}
+
+	for i := range npList.Items {
+		np := &npList.Items[i]
+		shouldReconcile := false
+
+		// Check ingress rules for named ports
+		for _, ingress := range np.Spec.Ingress {
+			for _, port := range ingress.Ports {
+				if port.Port != nil && port.Port.Type == intstr.String {
+					if podPortNames.Has(port.Port.StrVal) {
```

Comment:
What if the port name in Pod gets deleted, is it able to set shouldReconcile as true?

Reply:
good question, fixed in pod_handler
```
	if util.CheckPodHasNamedPort(*oldObj, "update") {
			return true
		}
		if util.CheckPodHasNamedPort(*newObj, "update") {
			return true
               }
		
```